python/taichi/lang/struct.py

```diff
@@ -222,10 +222,20 @@ def fill(self, val):
         Args:
             val (Union[int, float]): Value to fill.
         """
-        def assign_renamed(x, y):
-            return ops.assign(x, y)
 
-        return self._element_wise_writeback_binary(assign_renamed, val)
+        bcast_val = self._broadcast_copy(val)
```

Comment:
Is it possible to simply use `val`? It seems that `_element_wise_binary` will also broadcast.